axon-server-connector/src/main/java/org/axonframework/axonserver/connector/processor/EventProcessorController.java

```diff
@@ -16,60 +17,132 @@
 
-    private final EventProcessingConfiguration eventProcessingConfiguration;
+    private static final Logger logger = LoggerFactory.getLogger(EventProcessorController.class);
 
+    private final EventProcessingConfiguration eventProcessingConfiguration;
     private final Deque<Consumer<String>> pauseHandlers = new ArrayDeque<>();
-
     private final Deque<Consumer<String>> startHandlers = new ArrayDeque<>();
 
+    /**
+     * Instantiate a controller to perform operations on the {@link EventProcessor} instances contained in this
+     * application. This controller should provide functionality to trigger all the APIs available on the EventProcessor
+     * interface and it's implementations.
+     *
+     * @param eventProcessingConfiguration the {@link EventProcessingConfiguration} from which the existing
+     *                                     {@link EventProcessor}s will be retrieved
+     */
     public EventProcessorController(EventProcessingConfiguration eventProcessingConfiguration) {
         this.eventProcessingConfiguration = eventProcessingConfiguration;
     }
 
-    public EventProcessor getEventProcessor(String processorName){
-        return this.eventProcessingConfiguration
-                .eventProcessor(processorName)
-                .orElseThrow(() -> new RuntimeException("Processor not found"));
+    EventProcessor getEventProcessor(String processorName) {
+        return eventProcessingConfiguration.eventProcessor(processorName)
+                                           .orElseThrow(() -> new RuntimeException(
+                                                   "Processor [" + processorName + "] not found"
+                                           ));
+    }
+
+    void pauseProcessor(String processorName) {
+        getEventProcessor(processorName).shutDown();
+        this.pauseHandlers.forEach(consumer -> consumer.accept(processorName));
+    }
+
+    void startProcessor(String processorName) {
+        getEventProcessor(processorName).start();
+        this.startHandlers.forEach(consumer -> consumer.accept(processorName));
     }
 
-    public void pauseProcessor(String processor){
-        getEventProcessor(processor).shutDown();
-        this.pauseHandlers.forEach(consumer -> consumer.accept(processor));
+    void releaseSegment(String processorName, int segmentId) {
+        EventProcessor eventProcessor = getEventProcessor(processorName);
+        if (!(eventProcessor instanceof TrackingEventProcessor)) {
+            logger.info("Release segment requested for processor [{}] which is not a Tracking Event Processor");
+            return;
+        }
+        ((TrackingEventProcessor) eventProcessor).releaseSegment(segmentId);
     }
 
-    public void startProcessor(String processor){
-        getEventProcessor(processor).start();
-        this.startHandlers.forEach(consumer -> consumer.accept(processor));
+    void splitSegment(String processorName, int segmentId) {
```

Comment:
Errors are only logged here. When calling the split/merge from AxonServer, it should get information on the result.

Reply:
Yes, this was my main doubt too.
Discussed this with @saratry and @abuijze, and we figured for now not returning a result would be sufficient. If you believe a quick solution is possible in this scenario, I'd very much would like to hear it.

Reply:
I've created issue #1001 to adjust the functionality to return the result of a split and merge back to Axon Server.